pkg/filepromoter/gzip.go

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package filepromoter
+
+import (
+	"compress/gzip"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+
+	"k8s.io/klog"
+)
+
+// minCompressSize is the minimum size for even trying to compress files
+const minCompressSize = 8 * 1024
+
+// minCompressionRatio is the ratio we must obtain to consider a file compressible
+const minCompressionRatio = 0.9
+
+// maybeGzip will try to gzip the file, and will return a temp file if compression was worthwhile
+// If it returns a tempfile, that tempfile should be deleted by the caller
+func maybeGzip(localFile string) (string, error) {
+	originalStat, err := os.Stat(localFile)
+	if err != nil {
+		return "", fmt.Errorf("error getting stat of %q: %v", localFile, err)
+	}
+
+	if originalStat.Size() < minCompressSize {
+		klog.V(2).Infof("file %s is too small to compress", localFile)
+		return "", nil
+	}
+
+	tmpfile, err := gzipToTempFile(localFile)
+	if err != nil {
+		return "", fmt.Errorf("error compressing file: %v", err)
+	}
+
+	removeTempfile := true
```

Comment:
Would it be cleaner to just split this function here (make maybeGzip return `tmpfile` here), and move the tempfile deletion elsewhere?

For example, you could move lines 53-70 to a function that validates the gzipped file (if it fails validation, delete and return empty string; otherwise return tmpfile).

I was just wondering if it was worthwhile to break up this function into smaller pieces.